File: Project/app/src/main/java/com/helpme/sign_up.java
package com.helpme;

import android.app.ProgressDialog;
import android.graphics.Typeface;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AutoCompleteTextView;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import Model.user;
import Model.userDB;

public class sign_up extends AppCompatActivity {
    Button registerButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sign_up);
        Typeface face = Typeface.createFromAsset(getAssets(), "Fonts/Nabila.ttf");
        ((TextView) findViewById(R.id.Logo)).setTypeface(face);
        registerButton = findViewById(R.id.register);
        registerButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                user newUser = new user(
                        ((AutoCompleteTextView) findViewById(R.id.eMail)).getText().toString(),
                        ((AutoCompleteTextView) findViewById(R.id.firstName)).getText().toString(),
                        ((AutoCompleteTextView) findViewById(R.id.midName)).getText().toString(),
                        ((AutoCompleteTextView) findViewById(R.id.lastName)).getText().toString(),
                        "M",
                        "",
                        ((AutoCompleteTextView) findViewById(R.id.password)).getText().toString(),
                        ((AutoCompleteTextView) findViewById(R.id.phoneNumber)).getText().toString(),
                        ((AutoCompleteTextView) findViewById(R.id.userName)).getText().toString(),
                        10
                );
                userDB dbUser = new userDB(sign_up.this);
                dbUser.addUser(newUser);
            }
        });
    }
}

File: Project/app/src/main/java/com/helpme/SignUp.kt
package com.helpme

import Model.Functionalities
import Model.profession
import Model.user
import android.graphics.Typeface
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.AutoCompleteTextView
import android.widget.TextView

class SignUp : AppCompatActivity() {
    private var controller = Functionalities(this);
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sign_up)
        val face = Typeface.createFromAsset(assets, "Fonts/Nabila.ttf");
        (findViewById<TextView>(R.id.Logo)).typeface = face;
    }

    fun btnRegisterClicked(view: View) {
        val newUser = user(
                (findViewById<(AutoCompleteTextView)>(R.id.eMail)).text.toString(),
                (findViewById<(AutoCompleteTextView)>(R.id.firstName)).text.toString(),
                (findViewById<(AutoCompleteTextView)>(R.id.midName)).text.toString(),
                (findViewById<(AutoCompleteTextView)>(R.id.lastName)).text.toString(),
                "Male",
                "",
                (findViewById<(AutoCompleteTextView)>(R.id.password)).text.toString(),
                (findViewById<(AutoCompleteTextView)>(R.id.phoneNumber)).text.toString(),
                (findViewById<(AutoCompleteTextView)>(R.id.userName)).text.toString(),
                10,
                null,
                null,
                null,
                profession()
        );
        controller.signUp(newUser);
    }
}
